storage/gcloud/aio/storage/storage.py

```diff
@@ -115,9 +126,113 @@ def __init__(self, project, service_file, token=None, session=None):
         resp.raise_for_status()
         return await resp.json()
 
-    async def download_as_string(self, bucket, object_name, session=None):
-        return await self.download(bucket, object_name,
-                                   params={'alt': 'media'}, session=session)
+    async def _upload_resumable(self, url: str, bucket: str, object_name: str,
+                                data, headers: dict = None,
+                                session: aiohttp.ClientSession = None, timeout: int = 120,
+                                upload_type_resumable_always: bool = False):
+        # https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload
+        async def initiate_upload_session(object_name: str, headers: dict, session: aiohttp.ClientSession):
+            params = {
+                'uploadType': 'resumable',
+            }
+
+            metadata = {'name': object_name}
+            metadata_str = json.dumps(metadata)
```

Comment:
Rather than casting this, you can just use the `dict` object directly with `session.post(..., json=metadata, ...)` instead of `session.post(..., data=metadata_str, ...)`.

Reply:
Does the `aiohttp` library not set the `content-length` header automatically? ie. https://github.com/aio-libs/aiohttp/blob/ffaef545a99d67c12b2000d07ce43d885a0cbf1a/aiohttp/client_reqrep.py#L430-L431